docs/RFCS/20220608_split_jobs_system_tables.md

```diff
@@ -0,0 +1,279 @@
+
+
+###### Upgrade Path for system.jobs
+
+Changes to the job control state, as jobs step through the control state machine, would need to start writing to both tables, along with job creation (writing to intake). A migration can then copy all existing jobs to the new control table in the appropriate state to backfill. Subsequently control queries can begin to query the new table instead, until an additional cluster version disables writes to the existing table, followed by a migration that drops it. Note that writes to job detail and progress are discussed in the job_info section below; this would be for control-related writes only.
+
+---
+
+### system.job_status_progress + system.job_status_message
+
+#### Description
+
+The job_status_progress and system.job_status_messages tables store human-readable information about a job's current and recent status and progress.
+
+These tables are not read by the jobs control system at any point, and should not be read by the job itself either; they are intended solely for human-consumption via SHOW JOBS and the DB Console. 
+
+Both tables have many rows per job: Each time a job updates its persisted status or progress it will insert a new row into the respective table with the new value, and inspection tools will scan for the latest row per job to find its current status/progress. 
+
+Both tables have the same primary key: job ID plus row creation time. However they are separate as their payloads are both of very different size -- a small number versus a large string -- and are updated at very different frequencies; a completion percentage or timestamp likely updates continuously for the entirely life of a job while execution status may change a fixed number of times per execution phase.
+
+#### Schema
+
+```
+    CREATE TABLE system.job_status_progress (
+    	job_id INT NOT NULL,
+    last_updated TIMESTAMPTZ,
+     	percent_complete FLOAT, # null if job does not have a defined completion percentage
+    	timestamp_complete DECIMAL, # set only for jobs that advance in time e.g. cdc or replication
+    	PRIMARY KEY (job_id, last_updated DESC)
+    )
+    CREATE TABLE system.job_status_message (
+    	job_id INT NOT NULL,
+    last_updated TIMESTAMPTZ,
+    	status STRING, # like v1's running_status, e.g. "waiting for leases" or "importing rows"
+    	detail STRING, # more info, reason, etc.
+    	PRIMARY KEY (job_id, last_updated DESC)
+    )
+```
+
+
+
+#### SHOW JOBS
```

Comment:
It would be useful to show the queries that will be performed on these tables in order to implement something like `SHOW JOBS`. You mention below that `SHOW JOBS` will be powered by the `crdb_internal.jobs` table, but what are the queries or table accesses powering that table?

Reply:
Fair. I was hoping to punt on many of the implementation nitty gritty, particularly for the new control stuff, initially; our asap priority is to add the new info table, just as an available opt-in tool for existing jobs to use if they want to, with the overall control overhaul being sketched here just to make sure it'll all fit together when we're done